### PR TITLE
Removal of warning from backend.py file

### PR DESCRIPTION
**Pull request by amitsrivastava78**
Removed the compiler warning where rate to be used in the dropout op

### CODE REVIEW

File: tensorflow/python/keras/backend.py [hunk 1]
```diff
@@ -4023,12 +4023,16 @@ def dropout(x, level, noise_shape=None, seed=None):
...
-  retain_prob = 1. - level
   if seed is None:
     seed = np.random.randint(10e6)
   # the dummy 1. works around a TF bug
   # (float32_ref vs. float32 incompatibility)
-  return nn.dropout(x * 1., retain_prob, noise_shape, seed=seed)
+  try:
+    retain_prob = 1. - level if level is not None else None
+  except TypeError:
+    raise ValueError("You must provide a rate to dropout.")
```
Comment (fchollet) [inline tensorflow/python/keras/backend.py]: This should be structured as:

```
if level is None:
   raise ValueError('None values not supported. Received argument `level=None` in `keras.backend.dropout`.')
```

**Review by fchollet — COMMENTED** [review]
Thank you for the PR.

File: tensorflow/python/keras/backend.py (hunk 1 above)
Comment (amitsrivastava78) [reply tensorflow/python/keras/backend.py]: @fchollet Updated the code as per your comments, please review.

**Comment by amitsrivastava78** [comment]
@fchollet & @hgadig , can you pls review the PR

**Comment by hgadig** [comment]
@fchollet  Could you please approve this PR.

**Comment by amitsrivastava78** [comment]
@fchollet & @hgadig , i have updated the code as per the comments,kindly check and if all ok  then kindly approve the PR.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 15 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @fchollet, @caisq, @pavithrasv: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/python/keras/backend.py [hunk 2]
```diff
@@ -4023,12 +4023,16 @@ def dropout(x, level, noise_shape=None, seed=None):
...
       A tensor.
   """
-  retain_prob = 1. - level
   if seed is None:
     seed = np.random.randint(10e6)
   # the dummy 1. works around a TF bug
   # (float32_ref vs. float32 incompatibility)
-  return nn.dropout(x * 1., retain_prob, noise_shape, seed=seed)
+  if level is None:
+    raise ValueError('None values not supported. Received argument '
```
Comment (pavithrasv) [inline tensorflow/python/keras/backend.py]: Can we add a unit test for this please?

**Comment by gbaned** [comment]
@amitsrivastava78  Did you get a chance to look on reviewer comments? Please let us know on the update. Also request you have a look on conflicts. Thanks!

**Comment by amitsrivastava78** [comment]
@gbaned , sorry for the delay in response,  I have checked the implementation and this seems to have changed, so this PR is no more valid, i am closing this PR.

Thanks alot for your support, will keep interacting with you on other PRs.

Regards
Amit
